src/integrationtest/java/com/ericsson/ei/utils/HttpExecutor.java

```diff
@@ -0,0 +1,81 @@
+package com.ericsson.ei.utils;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import java.io.IOException;
+
+
+public final class HttpExecutor {
+
+    private static HttpExecutor instance;
+    private CloseableHttpClient client = HttpClientBuilder.create().build();
+    private static final Logger LOGGER = LoggerFactory.getLogger(HttpExecutor.class);
+
+    private HttpExecutor() {
+        
+    }
+    
+    public static HttpExecutor getInstance() {
+        if(instance == null) {
+            instance = new HttpExecutor();
+        }
+        
+        return instance;
+    }
+    
+    /**
+     * Close existing HttpClient and create a new one.
+     *
+     * @throws IOException
+     */
+    public void recreateHttpClient() {
+        if (this.client != null) {
+            try {
+                this.client.close();
+            } catch (IOException e) {
```

Comment:
the logging here is not specific. Make it more clear that an exception occurred while closing httpclient